### PR TITLE
Docs: Plugin E2E - Update docs

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

Now that create-plugin will scaffold Playwright and grafana/plugin-e2e instead of Cypress based grafana/e2e, the docs needs to be updated accordingly. This PR does the following changes:
* there's a new `Migrating from grafana/e2e` article. This guide walks through how to manually install Playwright and plugin-e2e, how to migrate tests, setup CI and how to uninstall grafana/e2e. 
* the `Get started` guide is now focused on plugins that have been scaffolded with plugin-e2e setup. only walks through the steps needed to run the specs that were generated by create plugin. 
* removing all notices about plugin-e2e being in beta 
* adding notice about grafana/e2e being deprecated

Do not merge before [this](https://github.com/grafana/plugin-tools/pull/847) PR is merged. 

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/grafana-plugins-platform-team/issues/105

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 1]
```diff
@@ -164,25 +54,7 @@ If you want to start a specific version of Grafana, you can do that by specifyin
 GRAFANA_VERSION=10.1.6 npm run server
```
Comment (tolzhabayev) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:54]: minor nit but since you are on it: 10.4.1 would be better

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+sidebar_position: 90
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ScaffoldPluginE2InstallNPM from '@snippets/plugin-e2e-install.npm.md';
+import ScaffoldPluginE2EInstallPNPM from '@snippets/plugin-e2e-install.pnpm.md';
+import ScaffoldPluginE2EInstallYarn from '@snippets/plugin-e2e-install.yarn.md';
+
+The [`@grafana/e2e`](https://www.npmjs.com/package/@grafana/e2e) tool will be depcrecated when Grafana 11.0.0 is released, and the support for this package will be dropped when Grafana 11.1.0 is released. We recommened all plugin authors to migrate their end-to-end tests to use Playwright and `@grafana/plugin-e2e` instead of Cypress and `@grafana/e2e`.
```
Comment (tolzhabayev) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:22]:
```suggestion
The [`@grafana/e2e`](https://www.npmjs.com/package/@grafana/e2e) tool will be depcrecated when Grafana 11.0.0 is released, and the support for this package will be dropped. We recommend all plugin authors to migrate their end-to-end tests to use Playwright and `@grafana/plugin-e2e` instead of Cypress and `@grafana/e2e`.
```

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+- how to run Playwright tests in CI
+- how to uninstall Cypress and `@grafana/e2e`
+
+## Manually installing @grafana/plugin-e2e
+
+If you have a plugin that was not scaffolded with setup for `@grafana/plugin-e2e` and `@playwright/test`, you can follow these steps to set it up manually.
+
+### Step 1: Installing Playwright
+
+The `@grafana/plugin-e2e` tool extends Playwright APIs, so you need to have `@playwright/test` with a minimum version of 1.41.2 installed as a dev dependency in the `package.json` file of your plugin. Refer to the [Playwright documentation](https://playwright.dev/docs/intro#installing-playwright) for instruction on how to install. Make sure you can run the example tests that were generated during the installation. If the example tests ran successfully, you may go ahead and delete as they won't be needed anymore.
```
Comment (tolzhabayev) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:37]: go ahead and delete _them_

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+export default defineConfig<PluginOptions>({
+    ...
+});
+```
+
+You're now ready to start writing tests. The [Get started](../get-started.md) guide to learn how to run tests locally.
+
+## Migrating tests
+
+There's no tooling in place for migrating existing `@grafana/e2e` based Cypress tests to `@grafana/plugin-e2e` based Playwright test. You may checkout the following resources to get inspiration:
```
Comment (tolzhabayev) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:121]:
```suggestion
There's no tooling in place for automatically migrating existing `@grafana/e2e` based Cypress tests to `@grafana/plugin-e2e` based Playwright tests. This means you would have to convert your tests one by one or replace them with a new set of tests based on Playwright. You may checkout the following resources to get inspiration on how the playwright tests should be written:
```

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+- [Best practices around test isolation](./setup-resources.md#test-isolation)
+- `@grafana/plugin-e2e` exposes Grafana end-to-end selectors via the `Selectors` fixture, so you no longer need to use selectors exposed via the `@grafana/e2e-selectors` package. Read more about end-to-end selectors in the [Selecting UI elements](./selecting-ui-elements.md) guide.
+
+## Running tests in CI
+
+To run Playwright tests targeting multiple versions of Grafana in CI, use one of the example workflows in the [CI](./ci.md) guide.
+
+:::note
+
+Note that Grafana does not offer any way to run end-to-end tests targeting multiple versions of Grafana in other CI platforms such as Drone.
```
Comment (tolzhabayev) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:136]:
```suggestion
Note that Grafana does not offer any way to run end-to-end tests targeting multiple versions of Grafana in other CI platforms such as Drone or CircleCI.
```

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+:::
+
+## Uninstalling Cypress and @grafana/e2e
+
+Although we recommend moving from `@grafana/e2e` to `@grafana/plugin-e2e` in a timely manner, there's nothing preventing you from having the two side by side during a transitional phase.
+
+When all Cypress tests have been migrated, open the terminal and run the following scripts from within your local plugin development directory:
+
+### 1. Remove tests and config file
```
Comment (tolzhabayev) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:146]:
```suggestion
### 1. Remove Cypress tests and config file
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 7]
```diff
@@ -196,24 +68,15 @@ groupId="package-manager"
...
-groupId="package-manager"
-queryString="current-package-manager"
-/>
+Next we suggest you checkout the following guides:
 
-![e2e testing of multiple Grafana versions](/img/e2e-version-plugin-dependency.png)
+- [Selecting UI elements in end-to-end tests](./selecting-ui-elements.md)
+- [Configure the resources you'll need](./setup-resources.md)
+- [How to test data source plugins](./test-a-data-source-plugin/index.md)
+<!-- - [How to test panel plugins](./test-a-panel-plugin.md) -->
```
Comment (sunker) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:82]: The PR that adds this guide is not merged yet. Uncommenting to get the build to pass

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+In this guide you'll learn:
+
+- how to manually setup `@grafana/plugin-e2e` in your plugin
+- how to migrate tests
+- how to run Playwright tests in CI
+- how to uninstall Cypress and `@grafana/e2e`
+
+## Manually installing @grafana/plugin-e2e
+
+If you have a plugin that was not scaffolded with setup for `@grafana/plugin-e2e` and `@playwright/test`, you can follow these steps to set it up manually.
```
Comment (sympatheticmoose) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:33]: This sentence doesn't quite read right for me. I'm not sure its necessary in all honesty. If you want to keep something like it, maybe  could change to something like:

> Plugins scaffolded with `create-plugin` v4.5.0 and greater automatically include configuration for `@grafana/plugin-e2e` and `@playwright/test`. To add this configuration manually, follow these steps:

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+```
+
+You're now ready to start writing tests. The [Get started](./get-started.md) guide to learn how to run tests locally.
+
+## Migrating tests
+
+There's no tooling in place for automatically migrating existing `@grafana/e2e` based Cypress tests to `@grafana/plugin-e2e` based Playwright tests. This means you would have to convert your tests one by one or replace them with a new set of tests based on Playwright. You may checkout the following resources to get inspiration on how the playwright tests should be written:
+
+- [How to test data source plugins](./test-a-data-source-plugin/index.md)
+<!-- - [How to test panel plugins](./test-a-panel-plugin.md) -->
```
Comment (sympatheticmoose) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:124]: Something here is messing with the rendering
Screenshot 2024-04-05 at 14 03 11

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+export default defineConfig<PluginOptions>({
+    ...
+});
+```
+
+You're now ready to start writing tests. The [Get started](./get-started.md) guide to learn how to run tests locally.
+
+## Migrating tests
+
+There's no tooling in place for automatically migrating existing `@grafana/e2e` based Cypress tests to `@grafana/plugin-e2e` based Playwright tests. This means you would have to convert your tests one by one or replace them with a new set of tests based on Playwright. You may checkout the following resources to get inspiration on how the playwright tests should be written:
```
Comment (sympatheticmoose) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:121]:
```suggestion
There's no tooling in place for automatically migrating existing `@grafana/e2e` based Cypress tests to `@grafana/plugin-e2e` based Playwright tests. This means you would have to convert your tests one by one or replace them with a new set of tests based on Playwright. Refer to the following resources to get inspiration on how the playwright tests should be written:
```

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+
+- [How to test data source plugins](./test-a-data-source-plugin/index.md)
+<!-- - [How to test panel plugins](./test-a-panel-plugin.md) -->
+- [App plugin with example tests](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/app-basic/tests)
+- [Panel plugin with example tests](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/panel-datalinks/tests)
+- [Best practices around test isolation](./setup-resources.md#test-isolation)
+- `@grafana/plugin-e2e` exposes Grafana end-to-end selectors via the `Selectors` fixture, so you no longer need to use selectors exposed via the `@grafana/e2e-selectors` package. Read more about end-to-end selectors in the [Selecting UI elements](./selecting-ui-elements.md) guide.
```
Comment (sympatheticmoose) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:128]: This looks strange, either just remove from the bullets and have as its own paragraph. Or shorten to "How to select UI elements".

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+The `@grafana/plugin-e2e` package also exports a type named `PluginOptions` that you can use to extend the Playwright configuration with the `@grafana/plugin-e2e` specific options.
+
+```ts title="playwright.config.ts"
+import { defineConfig } from '@playwright/test';
+import type { PluginOptions } from '@grafana/plugin-e2e';
+
+export default defineConfig<PluginOptions>({
+    ...
+});
+```
```
Comment (jackw) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:115]: nit: I would combine this with the above playwright.config.ts snippet so it's a copy and paste once rather than copy and paste then manually update with this typing.

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+
+:::note
+
+Note that Grafana does not offer any supported way to run end-to-end tests targeting multiple versions of Grafana in other CI platforms such as Drone or CircleCI. But you can easily configure your CI to replicate what the referenced Github Action is doing as there is nothing specific that we do that can not be done in other CI systems.
+
+:::
+
+## Uninstalling Cypress and @grafana/e2e
+
+Although we recommend moving from `@grafana/e2e` to `@grafana/plugin-e2e` in a timely manner, there's nothing preventing you from having the two side by side during a transitional phase.
```
Comment (sympatheticmoose) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:142]: We could warn here about issues targeting versions greater than 11 using cypress

**Review by sympatheticmoose — APPROVED** [review]
Added some comments, but overall lgtm. I've not tested links or the actual steps (but don't think theres many new instructions if any as part of this)

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+### 2. Uninstall dependencies
+
+```shell
+npm uninstall --save-dev @grafana/e2e @grafana/e2e-selectors
+```
+
+### 3. Update scripts
+
+In the `package.json` file, remove the `e2e:update` script entirely and update the `e2e` the following way:
+<Tabs>
```
Comment (jackw) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:162]: I'm not sure we need these tabs. The script to run playwright for any of the package managers can be:

`"e2e": "playwright test",`

File: docusaurus/docs/get-started/folder-structure.md [hunk 15]
```diff
@@ -46,7 +45,8 @@ myorg-myplugin-datasource/
 │   ├── module.ts
 │   ├── plugin.json
 │   └── types.ts
-└── tsconfig.json
+├── tsconfig.json
+└── tests
```
Comment (jackw) [inline docusaurus/docs/get-started/folder-structure.md:49]: Does this need `playwright.config.ts` adding after `pkg`. 🤔

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md (hunk 9 above)
Comment (jackw) [inline docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:124]: I think this comment is causing excess whitespace with MDX:

image

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 16]
```diff
@@ -13,138 +13,28 @@ keywords:
...
+:::note
+The `create-plugin` tool started scaffolding basic setup for `@grafana/plugin-e2e` from version 4.5.0. If you already have a plugin and it was scaffolded with an earlier version of the `create-plugin` tool, you may follow the [migration](./migrate-from-grafana-e2e.md) guide for instruction on how to install and configure `@grafana/plugin-e2e` manually.
 :::
 
 ## Before you begin
 
 You need to have the following:
 
-- Grafana [plugin development environment](https://grafana.com/developers/plugin-tools/get-started/set-up-development-environment).
+- Grafana [plugin development environment](https://grafana.com/developers/plugin-tools/get-started/set-up-development-environment)
```
Comment (jackw) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:33]: nit: full stop consistency [is preferred](https://grafana.com/docs/writers-toolkit/write/style-guide/style-conventions/#unordered-lists).
```suggestion
- Grafana [plugin development environment](https://grafana.com/developers/plugin-tools/get-started/set-up-development-environment).
```

File: docusaurus/docs/e2e-test-a-plugin/get-started.md [hunk 17]
```diff
@@ -196,24 +68,15 @@ groupId="package-manager"
 queryString="current-package-manager"
 />
 
-### Step 7: Run tests in CI
+### Step 3: Setup CI
 
-We recommend using a CI workflow to run end-to-end tests to continuously check for breakages. The following Github workflow runs end-to-end tests against a range of Grafana versions for every PR in your Github repository.
+If you choose to add a Github workflow when you scaffolded the plugin, the Playwright end-to-end tests will be run in CI targeting the most recent version of Grafana. To take full advantage of plugin-e2e, it's recommended to follow the instructions in the [CI](./ci.md) guide to run Playwright end-to-end tests targeting all versions of Grafana that your plugin supports.
```
Comment (jackw) [inline docusaurus/docs/e2e-test-a-plugin/get-started.md:73]:
```suggestion
If you chose to add a Github workflow when you scaffolded the plugin, the Playwright end-to-end tests will be run in CI targeting the most recent version of Grafana. To take full advantage of plugin-e2e, it's recommended to follow the instructions in the [CI](./ci.md) guide to run Playwright end-to-end tests targeting all versions of Grafana that your plugin supports.
```

**Review by jackw — APPROVED** [review]
Nice work @sunker ! 👏 

Nothing blocking here IMHO but caught a few typos.

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 18]
```diff
@@ -0,0 +1,172 @@
...
+  ],
+});
+```
+
+The authenticated state is stored on disk with the following file name pattern: `<plugin-root>/playwright/.auth/<username>.json`.
+
+To prevent these files from being version controlled, you can add the following line to your `.gitignore` file:
+
+```shell title=".gitignore"
+/playwright/.auth/
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:103]: Yes good catch!

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md (hunk 9 above)
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:128]: Alright - I'll remove it entirely. Will open a new pr next week when the panel guide is merged.

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 19]
```diff
@@ -0,0 +1,172 @@
...
+
+In the `package.json` file, remove the `e2e:update` script entirely and update the `e2e` the following way:
+<Tabs>
+<TabItem value="npm" default>
+`"e2e": "npx playwright test",`
+</TabItem>
+<TabItem value="yarn" >
+`"e2e": "yarn playwright test",`
+</TabItem>
+<TabItem value="npx">
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:173]: I removed them 👍

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md (hunk 9 above)
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md]: Yep I've removed the commented line now

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md (hunk 13 above)
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:135]: I don't think it's necessary. We've already made it clear at the top of the page that the package won't be supported after G11.

File: docusaurus/docs/e2e-test-a-plugin/introduction.md [hunk 20]
```diff
@@ -12,12 +12,12 @@ keywords:
...
 ---
 
-# Introduction to plugin E2E testing
+# Introduction to plugin end-to-end testing
 
 [`@grafana/plugin-e2e`](https://www.npmjs.com/package/@grafana/plugin-e2e?activeTab=readme) is designed specifically for Grafana plugin developers. It extends [`@playwright/test`](https://playwright.dev/) capabilities with relevant fixtures, models, and expect matchers; enabling comprehensive end-to-end testing of Grafana plugins across multiple versions of Grafana. This package simplifies the testing process, ensuring your plugin is robust and compatible with various Grafana environments.
 
-:::warning
-The `@grafana/plugin-e2e` tool is still in beta and subject to breaking changes.
+:::danger
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/introduction.md:19]: Yeah I think that makes sense. I'll remove this. 👍

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md [hunk 21]
```diff
@@ -0,0 +1,172 @@
...
+import ScaffoldPluginE2EInstallPNPM from '@snippets/plugin-e2e-install.pnpm.md';
+import ScaffoldPluginE2EInstallYarn from '@snippets/plugin-e2e-install.yarn.md';
+
+The [`@grafana/e2e`](https://www.npmjs.com/package/@grafana/e2e) tool will be depcrecated when Grafana 11.0.0 is released, and the support for this package will be dropped. We recommend all plugin authors to migrate their end-to-end tests to use Playwright and `@grafana/plugin-e2e` instead of Cypress and `@grafana/e2e`.
+
+In this guide you'll learn:
+
+- how to manually setup `@grafana/plugin-e2e` in your plugin
+- how to migrate tests
+- how to run Playwright tests in CI
```
[the remark this answers is not in the document]
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:30]: The manual installation bits could be moved to its own article...but for now I'll just reorganize the headings a bit. We can come back to this later.

**Comment by sunker** [comment]
Changes applied. Thanks for feedback everyone!

File: docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md (hunk 13 above)
Comment (sympatheticmoose) [reply docusaurus/docs/e2e-test-a-plugin/migrate-from-grafana-e2e.md:139]: I think theres a difference between not being supported and not working entirely, people might just expect that no new features would be added and fixes not applied. But not a blocker by any sense
